modules/Emails/include/ListView/ListViewDataEmails.php

```diff
@@ -269,209 +191,817 @@ public function getListViewData(
+
+                    // if $params tell us to override all ordering
+                    if(!empty($params['overrideOrder']) && !empty($params['orderBy'])) {
+                        $order = $this->getOrderBy(strtolower($params['orderBy']), (empty($params['sortOrder']) ? '' : $params['sortOrder'])); // retreive from $_REQUEST
+                    }
+                    else {
+                        $order = $this->getOrderBy(); // retreive from $_REQUEST
+                    }
+
+                    // still empty? try to use settings passed in $param
+                    if(empty($order['orderBy']) && !empty($params['orderBy'])) {
+                        $order['orderBy'] = $params['orderBy'];
+                        $order['sortOrder'] =  (empty($params['sortOrder']) ? '' : $params['sortOrder']);
+                    }
+
+                    //rrs - bug: 21788. Do not use Order by stmts with fields that are not in the query.
+                    // Bug 22740 - Tweak this check to strip off the table name off the order by parameter.
+                    // Samir Gandhi : Do not remove the report_cache.date_modified condition as the report list view is broken
+                    $orderby = $order['orderBy'];
+                    if (strpos($order['orderBy'],'.') && ($order['orderBy'] != "report_cache.date_modified")) {
+                        $orderby = substr($order['orderBy'],strpos($order['orderBy'],'.')+1);
+                    }
+                    if ($orderby != 'date_entered' && !in_array($orderby, array_keys($filter_fields))) {
+                        $order['orderBy'] = '';
+                        $order['sortOrder'] = '';
+                    }
+
+                    if (empty($order['orderBy'])) {
+                        $orderBy = '';
+                    } else {
+                        $orderBy = $order['orderBy'] . ' ' . $order['sortOrder'];
+                        //wdong, Bug 25476, fix the sorting problem of Oracle.
+                        if (isset($params['custom_order_by_override']['ori_code']) && $order['orderBy'] == $params['custom_order_by_override']['ori_code'])
+                            $orderBy = $params['custom_order_by_override']['custom_code'] . ' ' . $order['sortOrder'];
+                    }
+
+                    if (empty($params['skipOrderSave'])) { // don't save preferences if told so
+                        $current_user->setPreference('listviewOrder', $order, 0, $this->var_name); // save preference
+                    }
+
+                    // If $params tells us to override for the special last_name, first_name sorting
+                    if (!empty($params['overrideLastNameOrder']) && $order['orderBy'] == 'last_name') {
+                        $orderBy = 'last_name '.$order['sortOrder'].', first_name '.$order['sortOrder'];
+                    }
+
+                    $data = array();
+
+                    $temp = clone $seed;
```

Comment:
This code comes from the parent class. It's used to copy the fields from the seed (SugarBean) to iterate through each row from the database result so that it can then populate the seed (SugarBean) by calling the methods of the SugarBean. The reason for the duplication is that I needed to customise the filter in order to support the imap fields.